[AG-13490] Fix(row-sorting): abs sorting order and docs

sdwvit opened the pull request:
addresses TC6, TC4, TC8 and others [AG-13490](https://ag-grid.atlassian.net/browse/AG-13490)
 
<img width="1021" height="255" alt="Screenshot 2025-12-01 at 17 34 11" src="https://github.com/user-attachments/assets/a7828276-7f46-4658-89c4-1ccda6cc4804" />

sdwvit committed:
Fix sorting index calculation in sort service

- Ensure proper wraparound to 0 when index exceeds array length

sdwvit committed:
Update sorting configuration to include sortingOrder for absolute sort

- Add sortingOrder to rankingChange column to enable absolute sort
- Clarify that sort property alone is insufficient for absolute sort
- Include note explaining the requirement of sortingOrder for absolute sort functionality

sdwvit committed:
better example

StephenCooper commented on documentation/ag-grid-docs/src/content/docs/row-sorting/index.mdoc:
```diff
@@ -139,13 +139,21 @@ In this example, the column `rankingChange` uses the following sort definition:
+        {
+            field: 'rankingChange',
+            sort: { direction: 'asc', type: 'absolute' },
+            sortingOrder: [null, { direction: 'asc', type: 'absolute' }, { direction: 'desc', type: 'absolute' }],
```
I am wondering if we should export constants for these sorts and the default sorting order. Might be a nice syntatatic sugar for this configuration.

sdwvit replied:
since default sorting order is built in now, and is applied if user doesn't specify a custom sortingOrder, do you still think they need it?

sdwvit committed:
Update sortingOrder defaults and clarify absolute sort requirements

- Set default sortingOrder to include absolute sort options when sort type is 'absolute'
- Update sortingOrder calculation to use available sort types from column definition
- Improve documentation by clarifying that sort property alone is insufficient for absolute sort
- Add note explaining the requirement of sortingOrder for absolute sort functionality

StephenCooper commented on packages/ag-grid-community/src/entities/agColumn.ts:
```diff
@@ -52,6 +52,11 @@ export function isColumn(col: Column | ColumnGroup | ProvidedColumnGroup): col i
 
 const DEFAULT_SORTING_ORDER: SortDirection[] = ['asc', 'desc', null];
+const DEFAULT_ABSOLUTE_SORTING_ORDER: (SortDef | SortDirection)[] = [
+    ...DEFAULT_SORTING_ORDER,
```
I would query if this is what I would expect to be the default if I only specified an absolute sort.

Thinking ahead to if we support case insensitive this same approach wouldn't make as much sense. 

So wondering if it is simpler to say the default will be asc and desc and null within the same sort type.

sdwvit replied:
good idea, I'll keep it till I work on case insensitive sort

sdwvit committed:
lint

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
apply qa comments

sdwvit committed:
Update sortingOrder defaults for absolute sort
